webserver/webserver/src/main/java/io/helidon/webserver/RouteListRoutingRules.java

```diff
@@ -110,6 +110,23 @@ private boolean isEmpty() {
         return records.isEmpty() && newWebServerCallbacks.isEmpty() && contextServices.isEmpty();
     }
 
+    @Override
+    public RouteListRoutingRules register(TracingConfiguration tracingConfiguration) {
+        onNewWebServer(ws -> ws.context().register(tracingConfiguration.envConfig()));
+
+        Service[] services = {tracingConfiguration.service()};
+        Record record = new Record(null, services);
+
+        // need tracing service to be the very first one, as it must start the span before other handlers are invoked
+        if (records.isEmpty()) {
```

Comment:
This can be reduced only to records.add(0, record). If you are adding element to the first index of empty List, it will work the same as if you call add(record).